examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -181,78 +183,96 @@ def best_low_complexity(cv_results):
 
 # Add a column to mark the selected model
 results_df = results_df.with_columns(
-    pl.when(pl.col('n_components') == best_components)
-    .then(pl.lit('Selected'))
-    .otherwise(pl.lit('Regular'))
-    .alias('model_type')
+    pl.when(pl.col("n_components") == best_components)
+    .then(pl.lit("Selected"))
+    .otherwise(pl.lit("Regular"))
+    .alias("model_type")
 )
 
 # Create a figure with two subplots: one for test scores and one for train vs
 # test comparison
 fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 6))
 
 # Plot 1: Test scores with error bars
-components = results_df['n_components'].to_numpy()
-colors = ['green' if comp == best_components else 'skyblue' for comp in components]
+components = results_df["n_components"].to_numpy()
+colors = ["green" if comp == best_components else "skyblue" for comp in components]
 
 bars = ax1.bar(
     components,
-    results_df['mean_test_score'].to_numpy(),
+    results_df["mean_test_score"].to_numpy(),
```

Comment:
are all these `to_numpy`s necessary? Polars Series implement `__array__` so it should be possible to just pass them in directly to matplotlib